src/Viz.js

```diff
@@ -296,6 +301,16 @@ export default class Viz extends BaseClass {
       dataNest.rollup(leaves => this._filteredData.push(merge(leaves, this._aggs))).entries(flatData);
 
     }
+    else if (this._noDataMessage) {
```

Comment:
This `else if` won't work, as we need to be checking on `this._filteredData.length`. 

Make this a plain ol' if statement with `this._noDataMessage && !this._filteredData.length` as the conditional.